### ISSUE TITLE
Terminal background is not used

### ISSUE THREAD

**Issue by Kaycell** [catppuccin/nvim#413]
### Description

👋🏼 
TL;DR: It seems the theme does not follow the `background` var as expected and always default to `mocha`.

I've tried with different package managers (Lazy and Plug) and terminals (Kitty and OSX default Terminal). When the term use a light theme, starting neovim with catppuccin defaults to the mocha flavor.
Shouldn't it use the `background` variable to set the appropriate flavor?
I don't experience this behavior with nvim default theme or other themes (Rose Pine or Tokyo Night for example)

### Neovim version

```markdown
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@Ventura-arm64.local
```

### Operating system and version

macOS 13.2

### Catppuccin version / branch / rev

fe32c776d86e8ba608e6294c43302c985835bfea

### Steps to reproduce

1. Open a terminal with a light theme (With Kitty: `kitty +kitten themes --reload-in=all Catppuccin-Latte`)
2. `nvim -nu NONE`
3. `:echo &background` => should output `light`
4. `nvim -nu minimal.lua` => Use the mocha flavor 🤔 
5. `:echo &background` => `dark` 🤔

### Expected behavior

In a light themed terminal (c.f: `nvim -u NONE` => `:echo &background` => `light`):
Catppuccin should use Latte or whatever flavor is set for "light"
In a dark themed terminal (c.f: `nvim -u NONE` => `:echo &background` => `dark`):
Catppuccin should use Mocha or whatever flavor is set for "dark"

### Actual behavior

Catppuccin doesn't care about the background var and seems to override it 🤔 

### Minimal config

```Lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/folke/lazy.nvim.git",
		"--branch=stable", -- latest stable release
		lazypath,
	})
end
vim.opt.rtp:prepend(lazypath)

require("lazy").setup({
	{
		lazy = false,
		"catppuccin/nvim",
		name = "catppuccin",
		config = function()
			require("catppuccin").setup()
  			vim.cmd("colorscheme catppuccin")
		end
	},
})
```

**Comment by zschaffer** [comment]
this looks like it still isn't fixed for some reason. 

Trying with the same minimal config on 128af65 gets the similar 'dark' auto set even when kitty's theme is catppuccin-latte with `:verbose set background` still stating that background=dark was set in `$CACHE/catppuccin/colors/catppuccin.vim line 1`

**Comment by nullchilly** [comment]
@zschaffer https://github.com/neovim/neovim/issues/22614 Maybe this is related?

**Comment by fdelacruz** [comment]
This is still broken and only works in NVIM v0.9.0 Release.

**Comment by nullchilly** [comment]
Update to latest neovim version (0.9.4)

I fixed it upstream in version 0.9.1

**Comment by fdelacruz** [comment]
<img width="719" alt="Screen Shot 2023-11-06 at 7 50 54 PM" src="https://github.com/catppuccin/nvim/assets/148098/4394608a-cc0f-467e-a1e6-f740d3257d29">

**Comment by fdelacruz** [comment]
It doesn't looks like its fixed onthe latest neovim version (0.9.4) yet..
<img width="614" alt="Screen Shot 2023-11-06 at 8 01 56 PM" src="https://github.com/catppuccin/nvim/assets/148098/d1dd5cd8-4f0e-4f38-a573-62d11ae6e886">
